pkg/controller/transaction/controller.go

```diff
@@ -171,14 +171,31 @@ func (r *Reconciler) reconcileTransactionChangeValidating(ctx context.Context, t
 			return controller.Result{}, errors.NewNotFound("model plugin not found")
 		}
 
+		// validation should be done on the whole config tree if it does exist. the list of path values
+		// that can be used for validation should include both of current path values in the configuration and new ones part of
+		// the transaction
+		configID := configuration.NewID(targetID, change.TargetType, change.TargetVersion)
+		config, err := r.configurations.Get(ctx, configID)
+		if err != nil {
+			if !errors.IsNotFound(err) {
+				log.Errorf("Failed applying Transaction %d to target '%s'", transaction.Index, targetID, err)
+				return controller.Result{}, err
+			}
+		}
 		pathValues := make([]*configapi.PathValue, 0, len(change.Values))
+		currentConfigValues := make(map[string]*configapi.PathValue)
+		if config.GetValues() != nil {
```

Comment:
I think you can/should just use `config.Values`. The `GetValues()` method is really just there so you can define interfaces for your Protobuf generated types to implement.